src/ui/components/lineage/sidebarTabsSwitchers/DisplayLogs.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react';
+import styles from '../index.module.scss';
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
+import { okaidia } from 'react-syntax-highlighter/dist/esm/styles/prism';
+import { fetchStepLogs } from '../../../layouts/runs/RunDetail/sidebarServices';
+import { useSelector } from 'react-redux';
+import { sessionSelectors } from '../../../../redux/selectors';
+import { FullWidthSpinner } from '../../spinners';
+
+type DisplayLogsProps = { selectedNode: any };
+
+const DisplayLogs = ({ selectedNode }: DisplayLogsProps) => {
+  const authToken = useSelector(sessionSelectors.authenticationToken);
+  const [logs, setLogs] = useState('');
```

Comment:
Moved the entire logic to a new component, so we only fetch the logs and start polling, once the tab is opened.